core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -0,0 +1,502 @@
+
+            new_languages_for_voiceover = (
+                set(complete_translation_language_list) - set(
+                exp_opp_summary.language_codes_with_assigned_voice_artists)
+            )
+
+            language_codes_needing_voice_artists_set = set(
+                exp_opp_summary.language_codes_needing_voice_artists)
+            language_codes_needing_voice_artists_set |= set(
+                new_languages_for_voiceover)
+
+            exp_opp_summary.language_codes_needing_voice_artists = list(
+                language_codes_needing_voice_artists_set)
+
+            exp_opp_summary.validate()
+
+            with datastore_services.get_ndb_context():
+                exp_opp_summary_model = (
+                    opportunity_models.ExplorationOpportunitySummaryModel(
+                        id=exp_opp_summary.id,
+                        topic_id=exp_opp_summary.topic_id,
+                        topic_name=exp_opp_summary.topic_name,
+                        story_id=exp_opp_summary.story_id,
+                        story_title=exp_opp_summary.story_title,
+                        chapter_title=exp_opp_summary.chapter_title,
+                        content_count=exp_opp_summary.content_count,
+                        incomplete_translation_language_codes=(
+                            exp_opp_summary
+                                .incomplete_translation_language_codes),
+                        translation_counts=exp_opp_summary.translation_counts,
+                        language_codes_needing_voice_artists=(
+                            exp_opp_summary
+                                .language_codes_needing_voice_artists),
+                        language_codes_with_assigned_voice_artists=(
+                            exp_opp_summary
+                                .language_codes_with_assigned_voice_artists)
+                    )
+                )
+
+            datastore_services.update_timestamps_multi([exp_opp_summary_model])
+
+        return exp_opp_summary_model
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of results from the exploration migration.
+
+        Returns:
+            PCollection. A PCollection of results from the exploration
+            migration.
+        """
+
+        unmigrated_exploration_models = (
+            self.pipeline
+            | 'Get all non-deleted exploration models' >> (
+                ndb_io.GetModels(
+                    exp_models.ExplorationModel.get_all(
+                        include_deleted=False)))
+            # Pylint disable is needed because pylint is not able to correctly
+            # detect that the value is passed through the pipe.
+            | 'Add exploration keys' >> beam.WithKeys(  # pylint: disable=no-value-for-parameter
+                lambda exp_model: exp_model.id)
+            | 'Remove broken exploration' >> beam.Filter(
```

Comment:
What's the long-term plan for this part?

Also what happens to these explorations? They never get migrated? Would that cause issues? /cc @DubeySandeep

Reply:
After the explorations are fixed we can reenable this, I should add a TODO here, but I cannot find the issue for this rn, does it exist or do I need to create a new one?

Reply:
No idea, I would suggest creating a new one if you can't find it (maybe discuss with @DubeySandeep). But in general, better to have 2 issues than none at all.

Reply:
Done.